pkg/splunk/enterprise/monitoringconsole.go

```diff
@@ -147,6 +147,10 @@ func getMonitoringConsoleStatefulSet(client splcommon.ControllerClient, cr splco
 		labels[k] = v
 	}
 
+	emptyVolumeSource := corev1.VolumeSource{
+		EmptyDir: &corev1.EmptyDirVolumeSource{},
+	}
+
```

Comment:
It looks like this would happen for the MC if `spec.EphemeralStorage` was set to `true` in the common function `getSplunkStatefulSet()`.  Now since, as you pointed out, the MC doesn't have it's own CR, we cannot set this value for just the MC.  We could move this check to the common function:
```
func getSplunkStatefulSet(client splcommon.ControllerClient, cr splcommon.MetaObject, spec *enterprisev1.CommonSplunkSpec, instanceType InstanceType, replicas int32, extraEnv []corev1.EnvVar) (*appsv1.StatefulSet, error) {
...
	// update template to include storage for etc and var volumes
	if spec.EphemeralStorage || type == SplunkMonitoringConsole {
		// add ephemeral volumes to the splunk pod for etc and opt
		emptyVolumeSource := corev1.VolumeSource{
			EmptyDir: &corev1.EmptyDirVolumeSource{},
		}
		statefulSet.Spec.Template.Spec.Volumes = []corev1.Volume{
			{Name: "mnt-splunk-etc", VolumeSource: emptyVolumeSource},
			{Name: "mnt-splunk-var", VolumeSource: emptyVolumeSource},
		}
```
But it's probably best served in the MC specific function until a MC CRD is created.

Reply:
Hi @jryb, thanks a lot for your review. On top of this, at the moment the common function ```getSplunkStatefulSet()``` used by all the other resources is not called for the MC. For sure it would be nice to have a dedicated CRD for MC and to reuse the same code.